java/src/main/native/src/TableJni.cpp

```diff
@@ -262,6 +262,56 @@ JNIEXPORT jlongArray JNICALL Java_ai_rapids_cudf_Table_gdfReadParquet(
   CATCH_STD(env, NULL);
 }
 
+JNIEXPORT jlongArray JNICALL Java_ai_rapids_cudf_Table_gdfReadORC(
+    JNIEnv *env, jclass j_class_object, jobjectArray filter_col_names, jstring inputfilepath,
+    jlong buffer, jlong buffer_length) {
+  bool read_buffer = true;
+  if (buffer == 0) {
+    JNI_NULL_CHECK(env, inputfilepath, "input file or buffer must be supplied", NULL);
+    read_buffer = false;
+  } else if (inputfilepath != NULL) {
+    JNI_THROW_NEW(env, "java/lang/IllegalArgumentException",
+                  "cannot pass in both a buffer and an inputfilepath", NULL);
+  } else if (buffer_length <= 0) {
+    JNI_THROW_NEW(env, "java/lang/IllegalArgumentException", "An empty buffer is not supported",
+                  NULL);
+  }
+
+  try {
+    cudf::jni::native_jstring filename(env, inputfilepath);
+    if (!read_buffer && filename.is_empty()) {
+      JNI_THROW_NEW(env, "java/lang/IllegalArgumentException", "inputfilepath can't be empty",
+                    NULL);
```

Comment:
Nit: Seems like this check would be better placed up where the code is already checking if an input file path was specified when the buffer was not.

Reply:
Please let me know if this is acceptable?

`if (buffer == 0) {
    JNI_NULL_CHECK(env, inputfilepath, "input file or buffer must be supplied", NULL);
   cudf::jni::native_jstring check_filepath(env, inputfilepath);
    if (check_filepath.is_empty()) {
      JNI_THROW_NEW(env, "java/lang/IllegalArgumentException", "inputfilepath can't be empty",
                    NULL);
      read_buffer=false
`

And in try{
 `try {
   cudf::jni::native_jstring filename(env, inputfilepath);
   cudf::jni::native_jstringArray n_filter_col_names(env, filter_col_names);
`

Reply:
Yeah, I'd rather not fetch the string via the JNI interface twice.  We can leave this as-is.